### ISSUE TITLE
Consider new GUI framework

### ISSUE THREAD

**Issue by roaldarbol** [roaldarbol/bux#40]
Could be either Qt or something that runs in the browser like `streamlit`.

**Comment by roaldarbol** [comment]
It might be possible to migrate to Qt6, as it miiiiight be possible to get PySide6 bindings: https://stackoverflow.com/a/74797491/13240268

**Comment by roaldarbol** [comment]
Nah, I'll stay with `tkinter`. But I've started a proper refactoring of the code, so frontend is separated from backend functionality - would make potentially switching in the future a lot easier and maintenance too!

**Comment by BrianPugh** [comment]
not to throw a wrench into your plans, but also consider:

https://github.com/gradio-app/gradio
https://pynecone.io

**Comment by roaldarbol** [comment]
All input is highly appreciated! One thing I currently need is support for `multiprocessing` and (at the moment) OpenCV. It seems that [gradio doesn’t support mp](https://github.com/gradio-app/gradio/issues/513) and I can’t find anything on it Pynecone. Just throwing you a bone, if you’d be interested in helping with development it would be immensely helpful - but no pressure!🙈

**Comment by roaldarbol** [comment]
I really like the idea of a webapp though!

**Comment by BrianPugh** [comment]
I've got enough of my own projects 😄 , but always happy to give an extra set of eyes.

**Comment by roaldarbol** [comment]
Hehe, I kinda figured - if you never ask you’ll never know.😜

**Comment by roaldarbol** [comment]
Consider using [magicgui](https://pyapp-kit.github.io/magicgui/), which napari uses. Will make it easier to potentially implement a plugin structure and make individual components available in Jupyter Notebooks as well. Super interesting.
